test-data/unit/check-classes.test

```diff
@@ -1405,6 +1405,74 @@ class X(Nope): pass  # E: Name 'Nope' is not defined
 a, b = X()  # Used to crash here (#2244)
 
 
+-- Class-valued attributes
+-- -----------------------
+
+[case testClassValuedAttributesBasics]
+class A: ...
+class B:
+    a = A
+    bad = lambda: 42
```

Comment:
OT: Interestingly, a zero-arg lambda here is treated differently than one with arguments; the zero-arg lamda is apparently type-analyzed on the spot (so its type is `def () -> int`) while the one with args is not (and ends up with type `Any`).